notify/sns/sns.go

```diff
@@ -212,24 +241,42 @@ func (n *Notifier) createPublishInput(ctx context.Context, tmpl func(string) str
 	return publishInput, nil
 }
 
-func validateAndTruncateMessage(message string, maxMessageSizeInBytes int) (string, bool, error) {
+func validateAndTruncateMessage(message string, maxMessageSize int) (string, bool, error) {
 	if !utf8.ValidString(message) {
 		return "", false, fmt.Errorf("non utf8 encoded message string")
 	}
-	if len(message) <= maxMessageSizeInBytes {
+
+	// Check if truncation is needed based on the appropriate metric (bytes or runes).
+	isSMS := maxMessageSize == 1600
```

Comment:
I think this slightly changes the behavior right? The previous code didn't check for this. I don't think it's a bad change, but it might be better to make this a separate change.

Reply:
Thank you for the feedback. You're right, it's better to separate the new behavior from this one. I will completely revert this function to main one.